src/notation/internal/notationuiactions.h

```diff
@@ -0,0 +1,69 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_NOTATION_NOTATIONUIACTIONS_H
+#define MU_NOTATION_NOTATIONUIACTIONS_H
+
+#include "ui/iuiactionsmodule.h"
+#include "notationtypes.h"
+#include "notationactioncontroller.h"
+#include "modularity/ioc.h"
+#include "context/iuicontextresolver.h"
+#include "async/asyncable.h"
+
+namespace mu::notation {
+class NotationUiActions : public ui::IUiActionsModule, public async::Asyncable
+{
+    INJECT(notation, context::IUiContextResolver, uicontextResolver)
+public:
+
+    NotationUiActions(std::shared_ptr<NotationActionController> controller);
```

Comment:
Can you explain please, why you don't use DI here?

Reply:
To use DI, we need to introduce separate controller interfaces, for example, `INotationActionController`. But I don’t want to do this, because I don’t want someone from outside to have access to them and use them (instead of a dispatcher, or ui actions register, etc.), and this is also not needed, because these two classes are always in the same module, closely are connected with each other and solve one task.